gold/Gold_mem.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+// TODO: the mem is byte (not efficient maybe line base?)
+
+#include <algorithm>
+
+#include "Gold_data.hpp"
+#include "robin_hood.hpp"
+
+class Gold_mem {
+public:
+
+  Gold_mem(std::function<uint8_t(uint64_t)> gb)
+   : get_byte(gb) {
+  }
+
+  void st_perform(const Gold_data &st_data) {
+    st_data.each_chunk([this](uint64_t addr, const std::vector<uint8_t> &data) {
+      for(auto b_pos = 0u;b_pos<data.size(); ++b_pos) {
+        mem_byte[addr+b_pos] = data[b_pos];
+      }
+    });
+  }
+
+  void ld_perform(Gold_data &ld_data) {
```

Comment:
So `ld_data` is a space memory array itself and `ld_perform` copies the data from there to the array of this model?  C++ seems to aid in obscuring what's going on here.